### PR TITLE
Fix config issues and add integration test.

### PR DESCRIPTION
**Pull request by laurenzlong**
- Fixes https://github.com/firebase/firebase-tools/issues/327.
- Adds integration test script so future changes won't cause regression
- Added "silent" parameter to API enablement, and set it to true for functions:config:<action> so that extra messages do not get printed. This allows developers to do things like:
`firebase functions:config:get > configs.json`

### CODE REVIEW

File: lib/functionsConfig.js [hunk 1]
```diff
@@ -60,22 +60,24 @@ exports.getFirebaseConfig = function(projectId, instance) {
 };
 
 
+/* If you make changes to this function, run "node scripts/test-functions-config.js"
```
Comment (mbleigh) [inline lib/functionsConfig.js]: Nit: just use two `//` lines instead of `/* */`.

File: lib/functionsConfig.js [hunk 2]
```diff
@@ -60,22 +60,24 @@ exports.getFirebaseConfig = function(projectId, instance) {
...
+   to ensure that nothing broke. */
 exports.setVariablesRecursive = function(projectId, configId, varPath, val) {
   try {
     var parsed = JSON.parse(val);
-    if (!_.isPlainObject(parsed)) { // could be a number, etc.
-      return _setVariable(projectId, configId, varPath, parsed);
+    if (!_.isPlainObject(parsed)) {
+      // parsed is a number
```
Comment (mbleigh) [inline lib/functionsConfig.js]: Couldn't this also have been a quoted string? What happens if I do:

    firebase functions:config:set foo.bar='"baz"'

Comment (laurenzlong) [inline lib/functionsConfig.js]: Yep, true! (Added test) Edited comment to reflect that.

File: lib/functionsConfig.js (hunk 1 above)
Comment (laurenzlong) [inline lib/functionsConfig.js]: Done

**Pull request merged**